rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSourceDescriptor.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.resourcemanager.nodesource;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.ow2.proactive.resourcemanager.authentication.Client;
+
+
+/**
+ *
+ */
+public class NodeSourceDescriptor implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private String name;
+
+    private String infrastructureType;
+
+    private Object[] infrastructureParameters;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make "infrastructureParameters" transient or serializable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1948)

Reply:
You can make this array Serializable[]? So you better communicate the `implements Serializable` implicatations of the class for non-transient fields.

Reply:
This is somehow a big refactoring to do as these parameters come from the ResourceManager interface.

Reply:
Ok, this is addressed now, by using List<Serializable>.